### PR TITLE
Improve pageview_timestamp function and integrate into api calls.

### PR DESCRIPTION
**Pull request by jeremiahpslewis**
- Fixed date format bug (seems like the api calls were updated)
- Improved the pageview_timestamp function so that it validates character strings
- Integrated the pageview_timestamp function so that one can now use date objects when calling article_pageviews
- Added tests for pageview_timestamp

### CODE REVIEW

File: R/timestamps.R [hunk 1]
```diff
@@ -19,29 +19,28 @@
...
-  # If it's a date, it'll only be 6 chars long, so we have to add 4 more
-  # using "first" to determine what they should be.
-  if(is_date){
-    if(first){
-      return(paste0(timestamps, "0100"))
+  template <- "%Y%m%d"
+
+  if("character" %in% class(timestamps)){
+    timestamps <- strptime(timestamps, paste0(template, "%H"))
+    if(any(is.na(timestamps))) {
```
Comment (Ironholds) [inline R/timestamps.R]: use anyNA, it's vaster.

File: R/query.R [hunk 2]
```diff
@@ -37,25 +37,30 @@
...
+
+
   article <- gsub(x = article, pattern = " ", replacement = "_", fixed = TRUE)
   if(length(article) > 1){
     return(lapply(article, article_pageviews, project = project, platform = platform,
                   user_type = user_type, start = start, end = end, ...))
   }
+
   # Handle timestamps
+  start <- pageview_timestamps(start)
```
Comment (Ironholds) [inline R/query.R:49]: What's the purpose of these calls, here? Is it to validate the ideally-now-character-string'd timestamps?

**Comment by jeremiahpslewis** [comment]
It's to validate / convert, depending on if you pass a character string or a date object. This way you don't have to know the API's special date format in order to use it, but can also just throw in `start = Sys.Date()`, etc.

**Comment by jeremiahpslewis** [comment]
In the future, would suggest making `pageview_timestamps` an internal method...only use is for figuring out proper formatting for API call.

File: R/timestamps.R (hunk 1 above)
Comment (jeremiahpslewis) [inline R/timestamps.R]: :+1:

**Comment by Ironholds** [comment]
Ahh, gotcha! That makes a ton of sense :)

**Comment by Ironholds** [comment]
Thanks!

**Comment by jeremiahpslewis** [comment]
np! great to have an active maintainer willing to merge. :)

**Pull request merged**